src/components/Editor/index.js

```diff
@@ -522,7 +512,33 @@ class Editor extends PureComponent {
     this.pendingJumpLocation = null;
   }
 
+  setText(nextProps) {
+    if (!this.state.editor) {
+      return;
+    }
+
+    if (!nextProps.selectedSource) {
```

Comment:
i'm changing the logic here a bit, because i think it's safe to always set the text to "" regardless of whether we currently have a selected source